### PR TITLE
Fix some LGTM alerts

### PR DESCRIPTION
**Pull request by Alain-Bearez**
Some time ago I started to fix the LGTM alerts, grouping them by their type. 

https://lgtm.com/projects/g/apache/poi/alerts/?mode=tree

Many groups are still to be handled but these few fixes might be worth to be pushed to trunk already.

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by centic9** [comment]
CI is running at https://builds.apache.org/view/P/view/POI/job/POI-Github-PullRequests/67

File: src/java/org/apache/poi/hssf/record/FilePassRecord.java [hunk 1]
```diff
@@ -88,32 +88,36 @@ public FilePassRecord(RecordInputStream in) {
...
+                    ((CryptoAPIEncryptionVerifier)encryptionInfo.getVerifier()).write(bos);
+                    break;
+                default:
+                    throw new EncryptedDocumentException("not supported");
+            }
 
-        out.write(data, 0, bos.getWriteIndex());
+            out.write(data, 0, bos.getWriteIndex());
+        } catch (IOException ioe) {
+            // should never happen in practice
```
Comment (centic9) [inline src/java/org/apache/poi/hssf/record/FilePassRecord.java:119]: doing such a catch looks could swallow exceptions in the future, I'd rather wrap the checked exception in an unchecked one if we know this is not expected anytime, e.g. IllegalStateException, so we still see that it happens if it should still happen at some point later

File: src/java/org/apache/poi/hssf/record/ObjRecord.java [hunk 2]
```diff
@@ -171,24 +172,28 @@ public int getRecordSize() {
...
+				while (out.getWriteIndex() < expectedEndIx) {
+					out.writeByte(0);
+				}
+			} else {
+				out.write(_uninterpretedData);
 			}
-		} else {
-			out.write(_uninterpretedData);
+		} catch (IOException ioe) {
+			// should never happen in practice
```
Comment (centic9) [inline src/java/org/apache/poi/hssf/record/ObjRecord.java]: ditto, better to throw an unchecked exception for cases where exceptions are not expected at all

File: src/java/org/apache/poi/hssf/record/StandardRecord.java [hunk 3]
```diff
@@ -17,51 +17,61 @@ Licensed to the Apache Software Foundation (ASF) under one or more
...
+            out.writeShort(getSid());
+            out.writeShort(dataSize);
+            serialize(out);
+            if (out.getWriteIndex() - offset != recSize) {
+                throw new IllegalStateException("Error in serialization of (" + getClass().getName() + "): "
+                        + "Incorrect number of bytes written - expected " + recSize + " but got "
+                        + (out.getWriteIndex() - offset));
+            }
+        } catch (IOException ioe) {
+            // should never happen in practice
```
Comment (centic9) [inline src/java/org/apache/poi/hssf/record/StandardRecord.java:59]: I'd throw unchecked here as well

File: src/java/org/apache/poi/hssf/record/cont/ContinuableRecord.java [hunk 4]
```diff
@@ -49,19 +51,26 @@ protected ContinuableRecord() {
...
-		return out.getTotalSize();
+		int totalSize = 0;
+		try (LittleEndianByteArrayOutputStream leo =
+				new LittleEndianByteArrayOutputStream(data, offset)) {
+			ContinuableRecordOutput out = new ContinuableRecordOutput(leo, getSid());
+			serialize(out);
+			out.terminate();
+			totalSize = out.getTotalSize();
+		} catch (IOException ioe) {
+			// should never happen in practice
```
Comment (centic9) [inline src/java/org/apache/poi/hssf/record/cont/ContinuableRecord.java:72]: Once more a unchecked exception would not hide a potential exception

File: src/java/org/apache/poi/ss/util/CellRangeAddressList.java [hunk 5]
```diff
@@ -124,9 +125,16 @@ public static int getEncodedSize(int numberOfRanges) {
 
 	public int serialize(int offset, byte[] data) {
 		int totalSize = getSize();
-		serialize(new LittleEndianByteArrayOutputStream(data, offset, totalSize));
+		try (LittleEndianByteArrayOutputStream lebaos =
+				new LittleEndianByteArrayOutputStream(data, offset, totalSize)) {
+			serialize(lebaos);
+		}
+		catch (IOException ioe) {
+			// should never happen in practice
```
Comment (centic9) [inline src/java/org/apache/poi/ss/util/CellRangeAddressList.java:133]: another location where unchecked exception might be more appropriate

**Comment by centic9** [comment]
Are the additional synchronizations based on some actual problem? We have a defined threading guarantee which states that a single workbook cannot be used in multiple threads, so many places do not need synchronization because they are local to one workbook/instance anyway.

Adding additional synchronization just because of static checkers is often not ideal as it may affect performance and may not actually fix an actual problem at all.

**Comment by centic9** [comment]
ok to test

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/POI-Github-PullRequests/68/

**Comment by centic9** [comment]
test-failure is unrelated, seems a new test now depends on the src/documentation folder to contain the separate repository for documentation, which is not good.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/POI-DSL-Github-PullRequests/3/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/POI-DSL-Github-PullRequests/6/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/POI-DSL-Github-PullRequests/8/

**Comment by pjfanning** [comment]
Apologies @Alain-Bearez - I've caused a few conflicts for your PR. I forgot about this PR and had been trying to fix a few of the arithmetic issues myself (because I ran into a few in some recent changes). I've been using commons-math ArithmeticUtils because it has some useful checks for overflows.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/POI-DSL-Github-PullRequests/10/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/POI-DSL-Github-PullRequests/11/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/POI-DSL-Github-PullRequests/12/

File: src/java/org/apache/poi/hssf/record/ObjRecord.java [hunk 6]
```diff
@@ -171,24 +172,29 @@ public int getRecordSize() {
...
-
-			for (int i = 0; i < subrecords.size(); i++) {
-				SubRecord record = subrecords.get(i);
-				record.serialize(out);
-			}
-			int expectedEndIx = offset+dataSize;
-			// padding
-			while (out.getWriteIndex() < expectedEndIx) {
-				out.writeByte(0);
+		try (LittleEndianByteArrayOutputStream out =
```
Comment (pjfanning) [inline src/java/org/apache/poi/hssf/record/ObjRecord.java]: @Alain-Bearez could you remove the tabs in this block?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/POI-DSL-Github-PullRequests/15/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/POI-DSL-Github-PullRequests/23/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/POI-DSL-Github-PullRequests/26/

**Comment by Alain-Bearez** [comment]
applied as r1859596

**Pull request closed without merging**